### PR TITLE
Add Concurrent Storage Reference

### PR DESCRIPTION
**Pull request by CommanderFoo**
First draft of the reference.

Will wait for more info from temp channel before asking for review on this.

### CODE REVIEW

File: src/references/concurrent_storage.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+id: concurrent_storage_reference
+name: Concurrent Storage
+title: Concurrent Storage
+tags:
+  - Reference
+---
+
+# Concurrent Storage
+
+**Concurrent Storage** allows creators to wite data to a table for players or games concurrently. Meaning the games will have the ability to write data when the player is offline (or online but on a different server), or write data that is shared across game instances.
```
Comment (DoubleA-MG) [inline src/references/concurrent_storage.md]: **Concurrent Storage** allows creators to write* data

File: src/references/concurrent_storage.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+Concurrent shared player storage works similar to [Shared Storage](../references/shared_storage.md). The concurrent storage API can get and set, and share between the creators games. The difference with concurrent storage, is that the shared key can be updated while the player is offline, and across different game instances.
+
+### Concurrent Shared Creator Storage
+
+Concurrent shared creator storage allows for creators to get and set data to a shared key that is available across different game servers. For example, if a player on one server finds a special enemy (i.e. for an event), other players on different server instances can be notified and also have the enemy spawned.
+
+## Read and Write Limits
+
+There is no request limit For reads, but if they are requested a lot they will end up cached and might be few seconds out of date.
```
Comment (DoubleA-MG) [inline src/references/concurrent_storage.md]: There is no request limit for* reads

**Review by DoubleA-MG — CHANGES_REQUESTED** [review]

**Review by DoubleA-MG — APPROVED** [review]

**Review by coreslinkous — APPROVED** [review]

**Comment by CommanderFoo** [comment]
Got some additional info I need to add to this reference which I will do tomorrow morning.

- I need to go over the info from Tung and make sure I didn't miss anything.
- Need to add an info box about testing.

File: src/references/concurrent_storage.md [hunk 3]
```diff
@@ -80,20 +80,51 @@ To save data to a concurrent shared storage key, creators will need to use its *
...
 
 Concurrent shared creator storage allows for creators to get and set data to a shared key that is available across different game servers. For example, if a player on one server finds a special enemy (i.e. for an event), other players on different server instances can be notified and also have the enemy spawned.
 
+You can use Concurrent Shared Creator Storage to create features like:
+
+- Cross Game and In-Game global data.
+- Global player list across Servers and Games.
+- Global shared currency, items etc.
+- Auction house.
+- Persistent world
```
Comment (DoubleA-MG) [inline src/references/concurrent_storage.md]: missing period

File: src/references/concurrent_storage.md [hunk 4]
```diff
@@ -80,20 +80,51 @@ To save data to a concurrent shared storage key, creators will need to use its *
 
 Concurrent player storage works similar to [Persistent Storage](../references/persistent_storage.md). The concurrent storage API can get and set data for a player that is online, but also with the added benefit of getting and setting data while the player is offline, and on different game instances.
 
+You can use Concurrent Player Storage to create features like:
+
+- Offline / Cross Server Player to Player trading.
+- Offline / Cross Server Player vs Player base raid.
+- Offline / Cross Server Player 2 Player farm help and visit.
```
Comment (DoubleA-MG) [inline src/references/concurrent_storage.md]: Assuming  Player to Player and Player 2 Player are the same, id choose to use one of them for clarity

File: src/references/concurrent_storage.md [hunk 5]
```diff
@@ -80,20 +80,51 @@ To save data to a concurrent shared storage key, creators will need to use its *
...
+
 ### Concurrent Shared Creator Storage
 
 Concurrent shared creator storage allows for creators to get and set data to a shared key that is available across different game servers. For example, if a player on one server finds a special enemy (i.e. for an event), other players on different server instances can be notified and also have the enemy spawned.
 
+You can use Concurrent Shared Creator Storage to create features like:
+
+- Cross Game and In-Game global data.
+- Global player list across Servers and Games.
+- Global shared currency, items etc.
```
Comment (DoubleA-MG) [inline src/references/concurrent_storage.md]: Global shared currency, items, etc.

missing comma

File: src/references/concurrent_storage.md (hunk 4 above)
Comment (DoubleA-MG) [reply src/references/concurrent_storage.md]: You copy and pasted the example features so line 87 still needs to be fixed, similar to line 98

**Review by DoubleA-MG — APPROVED** [review]

**Pull request merged**